ttnn/cpp/ttnn/operations/ccl/ccl_common.cpp

```diff
@@ -713,22 +712,21 @@ std::vector<tt_xy_pair> RingReduceScatterWrappedTensorSlicer::create_worker_slic
     std::size_t max_slice_size_in_tiles = max_slice_size_in_pages;
 
     // Assign slices by assuming that the input tensor is flattened into a 1D Shape
-    std::size_t optim_worker_slice_len_tiles = ceil(total_num_tiles / num_workers); // Ceil so that the remainder worker will have a smaller slice
+    std::size_t optim_worker_slice_len_tiles = ((total_num_tiles - 1) / num_workers) + 1; // Ceil so that the remainder worker will have a smaller slice
```

Comment:
This should be a constant

Reply:
Also I don't know the allowed values of num_workers. If it is restricted to only being a fixed set of configs that are powers of 2 the divide can be optimized quite a bit (a lot of architectures make those sorts of constraints so just asking in case)

Reply:
Num workers can be 1 or more.